packages/iov-bcp-types/src/transactions.ts

```diff
@@ -44,131 +37,62 @@ export interface Amount {
   readonly tokenTicker: TokenTicker;
 }
 
-export interface ChainAddressPair {
-  readonly chainId: ChainId;
-  readonly address: Address;
-}
-
-export enum TransactionKind {
-  AddAddressToUsername,
-  Send,
-  /** @deprecated see SetNameTx */
-  SetName,
-  SwapOffer,
-  SwapCounter,
-  SwapClaim,
-  SwapTimeout,
-  RegisterBlockchain,
-  RegisterUsername,
-  RemoveAddressFromUsername,
-}
-
-export interface BaseTx {
+/** The basic transaction type all transactions should extend */
+export interface UnsignedTransaction {
+  /**
+   * The domain in which the concrete transaction is valid
+   *
+   * This should be used for type detection only and not be encoded somewhere.
+   * Right now we use "bns", "ethereum", "lisk" and "rise" but this could also
+   * be migrated to a Java-style package names like "io.lisk.mainnet" later on.
+   */
+  readonly domain: string;
   /** the chain on which the transaction should be valid */
   readonly chainId: ChainId;
   readonly fee?: Amount;
   readonly gasPrice?: Amount;
   readonly gasLimit?: Amount;
   // signer needs to be a PublicKey as we use that to as an identifier to the Keyring for lookup
   readonly signer: PublicKeyBundle;
-  readonly ttl?: TtlBytes;
-}
-
-export interface AddAddressToUsernameTx extends BaseTx {
-  readonly kind: TransactionKind.AddAddressToUsername;
-  /** the username to be updated, must exist on chain */
-  readonly username: string;
-  readonly payload: ChainAddressPair;
 }
 
-export interface SendTx extends BaseTx {
-  readonly kind: TransactionKind.Send;
+export interface SendTransaction extends UnsignedTransaction {
+  readonly kind: "send";
   readonly amount: Amount;
-  readonly recipient: RecipientId;
+  readonly recipient: Address;
   readonly memo?: string;
 }
 
-/**
- * Associates a simple name to an account on a weave-based blockchain.
- *
- * @deprecated will be dropped in favour of RegisterUsernameTx
- */
-export interface SetNameTx extends BaseTx {
-  readonly kind: TransactionKind.SetName;
-  readonly name: string;
+export function isSendTransaction(transaction: UnsignedTransaction): transaction is SendTransaction {
```

Comment:
This is good. If we have this, why not also `isSwapTransaction` or such?

Reply:
Well spotted. No other reason than me being lazy plus the compiler did not force me to create them. Adding this to the list and resolving this comment.